### ISSUE TITLE
[Due for payment 2025-02-19] [$250] Scan - "Delete" and "Replace" options are displayed for an admin which causes an error

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#50113]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.43-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** https://expensify.testrail.io/index.php?/tests/view/5035807
**Email or phone of affected tester (no customers):** sustinov@applausemail.com
**Issue reported by:** Applause - Internal Team

## Action Performed:
Prerequisite
Create a workspace and invite an employee to the workspace
Login as an owner and employee
1. As the employee navigate to the workspace chat and click on the + icon > Submit expense > Scan
2. Upload a receipt and finish the flow
3. After the expense is created navigate to the tab where you are logged in as the admin/owner
4. Click on the receipt thumbnail
5. Here notice that the "Delete" and "Replace" options are displayed
6. Click on "Replace" > Choose another receipt or take a photo of it > Upload
7. Click on "Delete" > Confirm > Notice that an error is thrown

## Expected Result:
The workspace admin cannot edit the receipt file so there should not be "Delete" and "Replace" options

## Actual Result:
The ‘Delete’ and ‘Replace’ options are displayed, but when ‘Delete’ is executed, an error occurs and when ‘Replace’ is executed, the receipt is replaced

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/user-attachments/assets/164a7d6d-600e-43b3-929c-a6811ec8c88a

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021843656533685450338</li>
        <li>Upwork Job ID: 1843656533685450338</li>
        <li>Last Price Increase: 2025-01-29</li>
<li>Automatic offers: </li>
<ul>
<li>Krishna2323 | Contributor | 105994347</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @OfstadC</details>

**Comment by lanitochka17** [comment]
@OfstadC  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by lanitochka17** [comment]
We think that this bug might be related to #wave-collect - Release 1

**Comment by MelvinBot** [comment]
This has been labelled "Needs Reproduction". Follow the steps here: https://stackoverflowteams.com/c/expensify/questions/16989

**Comment by OfstadC** [comment]
I've asked for retesting in #qa

**Comment by OfstadC** [comment]
Waiting on QA to retest

**Comment by OfstadC** [comment]
@kavimuru any update? 😃

**Comment by kavimuru** [comment]
Still reproduced, sorry I missed this one.

https://github.com/user-attachments/assets/12f469d3-397e-47da-a8c0-6d70154c5892

**Comment by OfstadC** [comment]
Friendly bump @Ollyws 😃

**Comment by tgolen** [comment]
@lanitochka17 I confirmed that the frontend functionality is correct. Can you please update this issue to only focus on fixing the errors that happen? If this needs to be handled internally, that's fine, but I'd like to get some more information for this such as the logs for the delete/replace requests that have the errors.

@OfstadC If you're able to reproduce this, could you please try to get the `requestID` for me following [this SO](https://stackoverflowteams.com/c/expensify/questions/16809)?

**Comment by tgolen** [comment]
Bump @lanitochka17 @OfstadC

**Comment by OfstadC** [comment]
Sorry @tgolen ! Was OoO - looking now 😃

**Comment by OfstadC** [comment]
@tgolen I was able to reproduce today

>jsonCode
: 
405
message
: 
"405 Cannot detach receipt from unowned transaction"
onyxData
: 
[]
requestID
: 
"8e109b54d92daccc-MSP"

**Comment by tgolen** [comment]
Awesome! Here are [the logs](https://www.expensify.com/_devportal/tools/logSearch/#sort=asc&size=10000&query=request_id%3A%228e109b54d92daccc-MSP%22%20AND%20timestamp%3A%5B2024-11-11T00%3A00%20TO%202024-11-12T23%3A59%5D) for that request. 

It looks like that error is a very old error (at least 7 years old). So, it looks like in order for us to allow that operation, we need to make an update to Auth. 

Can you please go [here](https://github.com/Expensify/App/issues/new/choose) and open up a new issue using the "Open an internal issue for a backend fix" template with all the relevant info? Then it will get assigned to an engineer to fix.

**Comment by OfstadC** [comment]
[Done](https://github.com/Expensify/App/issues/52345)! Thanks @tgolen for your help!

**Comment by OfstadC** [comment]
Jules is assigned to E/E issue 😃

**Comment by OfstadC** [comment]
No update - E/E issue is waiting on a volunteer

**Comment by OfstadC** [comment]
No update

**Comment by OfstadC** [comment]
Still onhold

**Comment by OfstadC** [comment]
Samesies

**Comment by rlinoz** [comment]
Hey update from https://github.com/Expensify/App/issues/52345

It looks like Replace is now working, so we should be good on that.

What is left is to remove the `Delete` operation for admins/managers since we don't want to allow them to delete receipts, so we can remove the HOLD here and look for proposals

cc: @tgolen @Ollyws

**Comment by OfstadC** [comment]
Thanks @rlinoz !

**Comment by OfstadC** [comment]
@Ollyws Please complete BZ checklist prior to payment date 😃 Thank you!

**Comment by OfstadC** [comment]
## Payment Summary

- @Ollyws due $250 via manual NewDot request
- @Krishna2323 [paid](https://www.upwork.com/nx/wm/workroom/39816069/overview) $250 via Upwork

**Comment by garrettmknight** [comment]
$250 approved for @Ollyws
